File: apps/web/types/certificate.ts
export interface Certificate {
  id: string;
  user_id: string;
  title: string;
  issuer: string | null;
  issue_date: string | null;
  url: string | null;
  created_at: string;
  updated_at: string;
}

export interface CertificateCreate {
  title: string;
  issuer?: string | null;
  issue_date?: string | null;
  url?: string | null;
}

export type CertificateUpdate = Partial<CertificateCreate>;

File: apps/api/app/schemas/certificate.py
"""Certificate schemas"""

from datetime import date, datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, Field


class CertificateBase(BaseModel):
    title: str = Field(..., max_length=255)
    issuer: Optional[str] = Field(None, max_length=255)
    issue_date: Optional[date] = None
    url: Optional[str] = Field(None, max_length=1000)


class CertificateCreate(CertificateBase):
    pass


class CertificateUpdate(BaseModel):
    title: Optional[str] = Field(None, max_length=255)
    issuer: Optional[str] = Field(None, max_length=255)
    issue_date: Optional[date] = None
    url: Optional[str] = Field(None, max_length=1000)


class CertificateResponse(CertificateBase):
    id: str
    user_id: str
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

